### PR TITLE
h2: harden against potential deadlock when frame exceeds rx buffer capacity

### PR DESCRIPTION
**Pull request by cmoyes-jump**
- Fix a potential deadlock in the H2 "all or nothing" frame consumption path: if a non-`DATA` frame's total size (header + payload) exceeds the rx buffer capacity, `rx_suppress` would wait forever for data that can never arrive, deadlocking the connection. Note that production buffers are currently sized properly, so this protects against a future misconfig.
- Add guard in `fd_h2_rx1` that detects this condition and issues `FD_H2_ERR_INTERNAL` instead of endlessly waiting and deadlocking.
- Size test/fuzz buffer to match `max_frame_size` and clamp `max_frame_size` to buffer capacity so the existing `FRAME_SIZE` check catches oversized frames before they reach the buffer guard.
- Fix fuzzer loop to handle the one-step gap between conn error detection (`SEND_GOAWAY`) and teardown (`GOAWAY` sent, conn reset).
- New unit tests in test_h2_conn.c covering buffer guard edge cases.

Previously, this invariant was only implicit. The H2 "all or nothing" path silently assumes `bufsz >= max_frame_size + 9`. Nothing enforced that. Today's production configs are fine (2MB buffer >> 16KB max frame). If someone later shrinks the buffer or bumps `max_frame_size`, the deadlock becomes active.

This hardening change makes the buffer/frame-size relationship an explicit, enforced contract (the clamp + the `FD_TEST` + the runtime guard), rather than just a coincidence.

### CODE REVIEW

File: src/waltz/h2/fd_h2_conn.c [hunk 1]
```diff
@@ -668,6 +668,11 @@ fd_h2_rx1( fd_h2_conn_t *            conn,
 
   /* Consume all or nothing */
   ulong const tot_sz = sizeof(fd_h2_frame_hdr_t) + frame_sz;
+  if( FD_UNLIKELY( tot_sz>rbuf_rx->bufsz ) ) {
+    /* Frame will never fit in the buffer */
+    fd_h2_conn_error( conn, FD_H2_ERR_INTERNAL );
+    return;
+  }
   if( FD_UNLIKELY( tot_sz>fd_h2_rbuf_used_sz( rbuf_rx ) ) ) {
```
[the remark this answers is not in the document]
Comment (cmoyes-jump) [reply src/waltz/h2/fd_h2_conn.c:686]: In practice, nobody would set the buffer so small.
